sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -299,21 +299,49 @@ private static Method findAnnotatedMethod(
     }
 
     // We need to be able to call it. We require it is public.
-    checkArgument(
+    errors.checkArgument(
         (first.getModifiers() & Modifier.PUBLIC) != 0, "%s must be public", format(first));
```

Comment:
Would it make sense to shift this to the method-specific reporter, and then just do "Must be public"? (same q for static)